File: examples/scripts/hello.py
#!/usr/bin/env python3
"""
Simple Python script example for deps run
"""
import sys
import platform

def main():
    print(f"Hello from Python {platform.python_version()}!")
    print(f"Platform: {platform.system()} {platform.machine()}")

    if len(sys.argv) > 1:
        print(f"Arguments: {' '.join(sys.argv[1:])}")

    return 0

if __name__ == "__main__":
    sys.exit(main())

File: examples/scripts/hello.ts
#!/usr/bin/env tsx
/**
 * Simple TypeScript script example for deps run
 * Requires tsx or ts-node: npm install -g tsx
 */

import * as os from 'os';

function main(): number {
    console.log(`Hello from Node.js ${process.version} with TypeScript!`);
    console.log(`Platform: ${os.type()} ${os.arch()}`);

    const args: string[] = process.argv.slice(2);
    if (args.length > 0) {
        console.log(`Arguments: ${args.join(' ')}`);
    }

    return 0;
}

process.exit(main());
